pkg/icingadb/ha.go

```diff
@@ -324,6 +335,32 @@ func (h *HA) realize(ctx context.Context, s *icingaredisv1.IcingaStatus, t *type
 	return nil
 }
 
+func (h *HA) realizeLostHeartbeat() error {
+	return retry.WithBackoff(
+		h.ctx,
+		func(ctx context.Context) error {
+			stmt := h.db.Rebind("UPDATE icingadb_instance SET responsible = ? WHERE id = ?")
+			_, err := h.db.ExecContext(ctx, stmt, "n", h.instanceId)
+
+			return internal.CantPerformQuery(err, stmt)
+		},
+		IsRetryable,
+		backoff.NewExponentialWithJitter(time.Millisecond*256, time.Second*3),
+		retry.Settings{
+			OnError: func(_ time.Duration, attempt uint64, err, lastErr error) {
+				if lastErr == nil || err.Error() != lastErr.Error() {
+					log := h.logger.Debugw
+					if attempt > 2 {
+						log = h.logger.Infow
+					}
+
+					log("Can't update instance. Retrying", zap.Error(err), zap.Uint64("retry count", attempt))
+				}
+			},
+		},
+	)
+}
```

Comment:
`realize()` can also write `responsible = 'n'`:
https://github.com/Icinga/icingadb/blob/af41705f115bda2cd4870b16af6a56c0545a379c/pkg/icingadb/ha.go#L271

So I'm not sure if this new function is needed at all. Why is it?

Reply:
I am sure that in this case a transaction is not needed at all.

Reply:
`realize()` has already enough purpose(s), complexity and LOC.